aten/src/ATen/native/quantized/cpu/fake_quantize_per_channel_affine.cpp

```diff
@@ -0,0 +1,141 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/native/TensorIterator.h>
+#include <ATen/native/cpu/Loops.h>
+
+/* FakeQuantize Op for PerChannelAffine quantization scheme */
+namespace at {
+namespace native {
+
+/* Fake-quantizes the 'inputs' tensor.
+Args:
+  X: Forward input tensor.
+  dY: Backward input tensor (_backward op only).
+  scale: scale of per tensor affine quantization
+  zero_point: zero_point of per tensor affine quantization
+  quant_min: minimum quantized value
+  quant_max: maximum quantized value
+  quant_delay: Count of global steps for which to delay the quantization.
+               See note below.
+  iter: The current quantization iteration used for `quant_delay`.
+Returns:
+  Quantized tensor (double dtype).
+
+Notes:
+  - quant_delay might be set to non-zero to help weights stabilize in the
+    beginning of the training.
+  - quantization range [quant_min, quant_max]
+*/
+Tensor fake_quantize_per_channel_affine_cpu(
+    const Tensor& self,
+    const Tensor& scale,
+    const Tensor& zero_point,
+    int64_t quant_min,
+    int64_t quant_max) {
+  TORCH_CHECK(self.scalar_type() == ScalarType::Float);
+  TORCH_CHECK(
+      quant_min <= quant_max,
+      "`quant_min` should be less than or \
+        equal to `quant_max`.");
+
+//      return self;
+/*  TORCH_CHECK(
+#      zero_point >= quant_min && zero_point <= quant_max,
+#      "`zero_point` must be between `quant_min` and `quant_max`.");
+*/
+  auto Y = at::empty_like(self);
+  for (int i = 0; i < self.size(0); i++)
+  {
+    auto input_slice = self.slice(0,i,i+1);
+    auto output_slice = Y.slice(0,i,i+1);
+
+    float sc = scale[i].item().toFloat();
```

Comment:
you could use TensorAccessor (it's faster as indexing creating a tensor). But it matters less because you have slicing above that creates tensors anyway